### PR TITLE
Add Pylint (#44)

### PR DESCRIPTION
**Pull request by NathanielChavdarov**

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -63,17 +63,16 @@ jobs:
           virtualenv "$venv"
           source "$venv/bin/activate"
           pip install mypy
-          mypy algorithms/
+          mypy algorithms/ .
```
Comment (zannen) [inline .github/workflows/python.yml]: You are running `mypy` on two directories:
* `algorithms/`
* `.` (the current directory) which also happens to contain `algorithms`

So you are running `mypy` on `algorithms` twice.

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -63,17 +63,16 @@ jobs:
...
-           pip install pylint 
-           pylint algorithms/ 
+      - name: Type-check with pylint 
+        run: |
+          tmpdir="$(mktemp -d)" 
+          venv="$tmpdir/venv" 
+          virtualenv "$venv" 
+          source "$venv/bin/activate" 
+          pip install pylint 
+          pylint algorithms/ .
```
Comment (zannen) [inline .github/workflows/python.yml]: Please run the bash script I put in #44.

You can do this either by including the script in the yaml file, or (better) by having a separate file containing the bash script, then running the script from CI.

File: .github/workflows/python.yml (hunk 1 above)
Comment (NathanielChavdarov) [reply .github/workflows/python.yml]: "." has been removed

File: .github/workflows/python.yml (hunk 2 above)
Comment (NathanielChavdarov) [reply .github/workflows/python.yml]: done

File: pylint.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+usages="$(jq --arg x usage "$jqcmd" .pylint.json)"
+
+max_fatal=0
+max_error=0
+max_warning=0
+max_refactor=0
+max_convention=20
+max_usage=0
+
+if [[ "$fatals" -gt "$max_fatal" -o "$fatals" -gt "$max_fatal" -o "$errors" -gt "$max_error" -o "$warnings" -gt "$max_warning" -o "$refactor" -gt "$max_refactor" -o "$conventions" -gt "$max_convention" ]] then
```
Comment (zannen) [inline pylint.sh]: `-o` --> `||`

Comment (NathanielChavdarov) [reply pylint.sh]: done

**Pull request merged**